### PR TITLE
[SPARK-27557][DOC] Add copy button to Python API docs for easier copying of code-blocks

### PR DESCRIPTION
**Pull request by sangramga**
## What changes were proposed in this pull request?

Add a non-intrusive button for python API documentation, which will remove ">>>" prompts and outputs of code - for easier copying of code.

For example: The below code-snippet in the document is difficult to copy due to ">>>" prompts
```
>>> l = [('Alice', 1)]
>>> spark.createDataFrame(l).collect()
[Row(_1='Alice', _2=1)]

```
Becomes this - After the copybutton in the corner of of code-block is pressed - which is easier to copy 
```
l = [('Alice', 1)]
spark.createDataFrame(l).collect()
```

image

## File changes
Made changes to python/docs/conf.py and copybutton.js - thus only modifying sphinx frontend and no changes were made to the documentation itself- Build process for documentation remains the same.

copybutton.js -> This JS snippet was taken from the official python.org documentation site. 

## How was this patch tested?
NA

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Hm, I didn't know it needed that much of codes - I expected it's a configuration to switch on and off. Is there simpler way to do it? BTW where did `copybutton.js` code come from?

**Comment by sangramga** [comment]
@HyukjinKwon there seems to be no default configuration in Sphinx - Nature theme, also above seems to be the simplest way.
I referred to Sphinx issues: [https://github.com/rtfd/sphinx_rtd_theme/issues/167](https://github.com/rtfd/sphinx_rtd_theme/issues/167)
It's also most adopted in python community - refer to both scipy and scikit-learn documentation
[scipy-copybutton.js](https://github.com/scipy-lectures/scipy-lectures.github.com/blob/master/_static/copybutton.js)
[sklearn-copybutton.js](https://github.com/scikit-learn/scikit-learn/blob/master/doc/themes/scikit-learn/static/js/copybutton.js)

File: python/docs/_static/copybutton.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+// Copyright 2014 PSF. Licensed under the PYTHON SOFTWARE FOUNDATION LICENSE VERSION 2
```
Comment (srowen) [inline python/docs/_static/copybutton.js]: This license is compatible with ALv2. For completeness, I'd add an entry at the end of `LICENSE` noting the inclusion of software under this license, and add a file in `licenses/` containing the text of the license.

File: python/docs/_static/copybutton.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+// File originates from the cpython source found in Doc/tools/sphinxext/static/copybutton.js
```
Comment (srowen) [inline python/docs/_static/copybutton.js]: Given its source, is there any way to get this to render automatically with sphinx or anything? not sure that even makes sense, just checking if this is something that has to be copied into the project or not.

**Review by srowen — CHANGES_REQUESTED** [review]

File: python/docs/_static/copybutton.js (hunk 1 above)
Comment (sangramga) [reply python/docs/_static/copybutton.js]: @srowen Made the requested commits. Please check whether the LICENSE entries are appropriate.

File: python/docs/_static/copybutton.js (hunk 2 above)
Comment (sangramga) [reply python/docs/_static/copybutton.js]: @srowen I didn't find any other way to get this rendered with sphinx. According to my [references](https://github.com/apache/spark/pull/24456#issuecomment-486600616), it is used in the same way by other libraries like scipy, and scikit-learn.

**Review by srowen — APPROVED** [review]

File: licenses/LICENSE-copybutton.txt [hunk 3]
```diff
@@ -0,0 +1,254 @@
+A. HISTORY OF THE SOFTWARE
```
Comment (felixcheung) [inline licenses/LICENSE-copybutton.txt]: this looks like license for python?

**Review by felixcheung — COMMENTED** [review]
I have seen different versions of this and the license is just the section of `PYTHON SOFTWARE FOUNDATION LICENSE VERSION 2` under **B**

File: licenses/LICENSE-copybutton.txt (hunk 3 above)
Comment (sangramga) [reply licenses/LICENSE-copybutton.txt]: Yes, this is the license for python. Do you want me to just put `PYTHON SOFTWARE FOUNDATION LICENSE VERSION 2` under B in the licenses file?

Comment (srowen) [reply licenses/LICENSE-copybutton.txt]: I'm OK with trimming it down to the 'section B' part as I agree that the rest is not licensing terms. You can modify the heapq license file too accordingly.

**Comment by HyukjinKwon** [comment]
ok to test

**Review by HyukjinKwon — APPROVED** [review]
ok by me too

**Comment by srowen** [comment]
Merged to master

**Comment by srowen** [comment]
Merged to master

**Comment by gatorsmile** [comment]
@sangramga Could you also help us to do it in [SQL docs](https://spark.apache.org/docs/2.4.3/)?

**Pull request closed without merging**